### PR TITLE
Key rotation e2e uses the OSA RP

### PR DESCRIPTION
**Pull request by kargakis**
/cc charlesakalugwu 

Fixes https://github.com/openshift/openshift-azure/issues/807

### CODE REVIEW

**Comment by kargakis** [comment]
Turns out we can get rid of UpdateOSACluster

File: test/e2e/specs/fakerp/keyrotation.go [hunk 1]
```diff
@@ -53,21 +52,15 @@ var _ = Describe("Key Rotation E2E tests [KeyRotation][Fake][LongRunning]", func
...
-		Expect(err).NotTo(HaveOccurred())
-
 		By("Persisting the config blob containing the new certificates and credentials")
 		err = saveConfig(mutated, *configBlob)
 		Expect(err).NotTo(HaveOccurred())
 
 		By("Calling update on the fake rp with the updated config blob")
-		updated, err := cli.UpdateOSACluster(ctx, external, pluginConfig)
+		external.Properties.ProvisioningState = nil
+		updated, err := cli.OpenShiftManagedClusters.CreateOrUpdateAndWait(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), *external)
```
Comment (charlesakalugwu) [inline test/e2e/specs/fakerp/keyrotation.go]: this should be using `OpenShiftManagedClustersAdmin`

Comment (kargakis) [reply test/e2e/specs/fakerp/keyrotation.go]: Not before https://github.com/openshift/openshift-azure/pull/821 gets merged and also #821 is blocked on this PR so we cannot wait for it. :)

**Comment by kargakis** [comment]
/skip 

works fine locally up to the point of the compute isue

**Pull request merged**